de-lib/src/main/java/org/iplantc/de/analysis/client/views/ViceLogsView.java

```diff
@@ -0,0 +1,63 @@
+package org.iplantc.de.analysis.client.views;
+
+import org.iplantc.de.analysis.client.AnalysesView;
+import org.iplantc.de.analysis.client.ReactAnalyses;
+import org.iplantc.de.commons.client.util.CyVerseReactComponents;
+
+import com.google.gwt.user.client.ui.HTMLPanel;
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.Widget;
+
+import javax.inject.Inject;
+
+
+/**
+ * @author sriram
+ */
+public class ViceLogsView implements IsWidget {
+
+    HTMLPanel panel;
+    private ReactAnalyses.ViceLogsProps viceLogsProps;
+
+    @Inject
+    public ViceLogsView() {
+        panel = new HTMLPanel("<div></div>");
+        viceLogsProps = new ReactAnalyses.ViceLogsProps();
+        viceLogsProps.logs = "";
+    }
+
+    public void load(AnalysesView.Presenter presenter,
+                     String analysisName,
+                     String logs,
+                     String baseDebugId) {
+        viceLogsProps.logs = viceLogsProps.logs.concat(logs);
```

Comment:
I may not be following the logic correctly, but it seems like if the user initially follows the logs and then decides to hit `Refresh`, the endpoint will return all the logs because the `viceLogsSinceTime` will be set to 0. With this `concat` of the old logs and the newly fetched logs, it seems like they'll have the logs twice.

Reply:
Bug indeed. I also found another bug when users decides to follow logs, then closes the log dialog and then re-opens its follow logs switch will be `on`  but nothing happens. I found a fix. Will update the code.